### PR TITLE
Add support for containerd to kubecluster script

### PR DESCRIPTION
**Pull request by nagiesek**

### CODE REVIEW

**Comment by nagiesek** [comment]
Opened this for review, will update the repo locations to download from here instead of my own private repo before merging.

File: Kubernetes/windows/helper.v2.psm1 [hunk 1]
```diff
@@ -407,7 +413,12 @@ function DownloadCniBinaries($NetworkMode, $CniPath)
     DownloadFile -Url  "https://github.com/$Global:GithubSDNRepository/raw/$Global:GithubSDNBranch/Kubernetes/flannel/l2bridge/cni/flannel.exe" -Destination $CniPath\flannel.exe
     DownloadFile -Url  "https://github.com/$Global:GithubSDNRepository/raw/$Global:GithubSDNBranch/Kubernetes/flannel/l2bridge/cni/host-local.exe" -Destination $CniPath\host-local.exe
 
-    if ($NetworkMode -eq "l2bridge")
+    if ($Global:Cri -eq "containerd")
+    {
+        DownloadFile  "https://github.com/microsoft/windows-container-networking/releases/download/v0.2.0/windows-container-networking-cni-amd64-v0.2.0.zip" -Destination "$env:TEMP\windows-container-networking-cni-amd64-v0.2.0.zip"
+        Expand-Archive -Path "$env:TEMP\windows-container-networking-cni-amd64-v0.2.0.zip" -DestinationPath $CniPath -Force
+    }
+    elseif ($NetworkMode -eq "l2bridge")
```
Comment (madhanrm) [inline Kubernetes/windows/helper.v2.psm1]: Todo: This check should do the same as previous - if docker, then download it from GitHub.com/containernetworking/plugins repo

File: Kubernetes/windows/helper.v2.psm1 [hunk 2]
```diff
@@ -799,6 +915,11 @@ function GetKubeletArguments()
         $kubeletArgs += "--feature-gates=$KubeletFeatureGates"
     }
 
+    if ($IsContainerd) 
+    {
+       $kubeletArgs += @("--container-runtime=remote", "--container-runtime-endpoint=npipe:////./pipe/containerd-containerd")
```
Comment (madhanrm) [inline Kubernetes/windows/helper.v2.psm1]: Can you keep one arg in a line - Just for readability

File: Kubernetes/windows/kubeadm/KubeCluster.ps1 [hunk 3]
```diff
@@ -142,8 +142,11 @@ if ((Test-Path env:GITHUB_SDN_BRANCH) -and ($env:GITHUB_SDN_BRANCH -ne ''))
     $_GithubSDNBranch = $env:GITHUB_SDN_BRANCH
 }
 
-LoadPsm1 -Path "https://raw.githubusercontent.com/$_GithubSDNRepository/$_GithubSDNBranch/Kubernetes/windows/helper.v2.psm1"
-LoadPsm1 -Path "https://raw.githubusercontent.com/$_GithubSDNRepository/$_GithubSDNBranch/Kubernetes/windows/hns.psm1"
+LoadPsm1 -Path "c:\helper.v2.psm1"
```
Comment (nagiesek) [inline Kubernetes/windows/kubeadm/KubeCluster.ps1]: This will be reverted before merge

**Comment by madhanrm** [comment]
@PatrickLang  FYI

**Comment by madhanrm** [comment]
Looks good, will merge after we get the validation done.

**Comment by madhanrm** [comment]
Please update ReadMe to include this CRI. Also mention where you are picking the sources for containerd from.

Please add 2 sample configuration for containerd-flannel-vxlan.json and containerd-flannel-l2bridge.json

File: Kubernetes/windows/kubeadm/KubeCluster.ps1 (hunk 3 above)
Comment (madhanrm) [reply Kubernetes/windows/kubeadm/KubeCluster.ps1]: Is this ready for merge?

Comment (nagiesek) [reply Kubernetes/windows/kubeadm/KubeCluster.ps1]: Let me update. I'll upload these into this repo first

File: Kubernetes/windows/helper.v2.psm1 (hunk 1 above)
Comment (nagiesek) [reply Kubernetes/windows/helper.v2.psm1]: Does what is advertised.

File: Kubernetes/windows/helper.v2.psm1 (hunk 2 above)
Comment (nagiesek) [reply Kubernetes/windows/helper.v2.psm1]: Can you clarify?

**Pull request merged**